libtenzir/src/tql2/eval_impl.cpp

```diff
@@ -113,7 +116,8 @@ auto evaluator::eval(const ast::field_access& x) -> series {
   auto& s = caf::get<arrow::StructArray>(*l.array);
   for (auto [i, field] : detail::enumerate<int>(rec_ty->fields())) {
     if (field.name == x.name.name) {
-      return series{field.type, s.field(i)};
+      // TODO: Emit a warning if the record itself was `null`?
```

Comment:
The reason why I hesitate here is that I'm not sure how useful this warning would be. Unlike with many of the other "this thing that you are looking for doesn't exist", you can make a good case here that you'd almost always want to propagate the `null`. For example, we know that the field would exist if the parent record didn't happen to be `null`, so it's not "as bad" as accessing a field that could not ever exist for that type. But if you have a hunch that a warning is the correct behavior, I'll add it.

Reply:
My reasoning for wanting to have this is very simple: The type of `null` should never be observable by the user.

Reply:
Right, I agree that's something to strive for in general. But this PR changes field accesses to not print a warning for `null_type`. So with your argument, it should also not print a warning for `null: record_type` (unless you consider warnings not being part of the semantics, then this argument doesn't apply).

I think the problem is that we want to avoid warning cascades, but because we use `null_type` as our return type in such cases, we have to balance this with (a) true `null: null_type` values that were not created by a warning, and (b) `null` of other values which generally should behave the same. 

I can see us coming back to that "error" value that you suggested quote some time ago, albeit in slightly different form, where we use it as "null but please don't complain if you fail because I'm null". But I guess that requries some more pondering.

Reply:
Although, if that's our goal, then emitting a warning here is correct, because we didn't warn for this previously... I'll add a warning then, and in the more distant future we can reconsider :+1: